### PR TITLE
refactor: intercepting routes preparation

### PR DESCRIPTION
**Pull request by altaywtf**
https://github.com/vercel/next.js/issues/53170

### CODE REVIEW

**Comment by altaywtf** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #42** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/42?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #43** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/43?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #47** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/47?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #48** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/48?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #49** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/49?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #50** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/50?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
              * **PR #51** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/51?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #44** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/44?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #45** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/45?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/experiment-station/beecast/50?utm_source=stack-comment).

**Comment by altaywtf** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #42** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/42?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #43** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/43?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #47** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/47?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #48** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/48?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #49** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/49?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #50** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/50?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
              * **PR #51** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/51?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #44** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/44?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #45** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/45?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/experiment-station/beecast/50?utm_source=stack-comment).

File: app/@modal/default.tsx
Comment (altaywtf) [inline app/@modal/default.tsx:1]: not sure if we 100% need this, saw it on https://github.com/vercel-labs/nextgram/blob/main/src/app/%40modal/default.tsx

Comment (altaywtf) [reply app/@modal/default.tsx:1]: ok, we do 👍 otherwise everything is 404 👯

**Review by okanisildar — APPROVED** [review]

**Pull request merged**